### PR TITLE
docs: add google cloud storage bucket guide

### LINKED ISSUE #5541: Add support for Google Cloud Storage buckets in media backend
### Is your feature request related to a problem? Please describe.

It would be nice of Google Cloud Storage Buckets could be used for storing media data.

### Describe the solution you'd like

A way to use GCS Buckets in HedgeDoc.

### Describe alternatives you've considered

- Using a persistent volume to store, but this does not scale well.
- Setting up Minio and proxying traffic to GCS, but this would unnecessarily complicate the stack needed to run HedgeDoc on GCP

### Additional context

There is an official [Node.js Client library for Google Cloud Storage](https://cloud.google.com/nodejs/docs/reference/storage/latest) available which can be used.

### PR DESCRIPTION
**Pull request by syphernl**
### Component/Part
Documentation

### Description
This PR adds a guide on how to setup the S3 integration with Google Cloud Storage

### Steps

- [ ] Added implementation
- [ ] Added / updated tests
- [X] Added / updated documentation
- [ ] Added changelog snippet
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### Related Issue(s)
Closes #5541

### CODE REVIEW

**Comment by DerMolly** [comment]
Hey @syphernl,
thanks for your contribution!

In order to get this merged, we need you to sign-off your commits according to the contribution guidelines for the HedgeDoc organization. You can find more information on this here: https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md#certificate-of-origin

File: docs/content/guides/s3-google-cloud-storage.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+## Configure HedgeDoc
+
+In your `.env` file set:
+
+```env
+HD_MEDIA_BACKEND="s3"
+HD_MEDIA_BACKEND_S3_ENDPOINT="https://storage.googleapis.com"
+HD_MEDIA_BACKEND_S3_ACCESS_KEY="GOOG1EXXXXXXXXXX"
+HD_MEDIA_BACKEND_S3_SECRET_KEY="XXXXXXXXXX"
+HD_MEDIA_BACKEND_S3_BUCKET="your-bucket-name"
```
Comment (ErikMichelson) [inline docs/content/guides/s3-google-cloud-storage.md:73]:
```suggestion
Set the following environment variables (or the [corresponding config.json values](/configuration/#amazon-s3)):

```env
CMD_IMAGE_UPLOAD_TYPE="s3"
CMD_S3_ENDPOINT="https://storage.googleapis.com"
CMD_S3_ACCESS_KEY_ID="GOOG1EXXXXXXXXXX"
CMD_S3_SECRET_ACCESS_KEY="XXXXXXXXXX"
CMD_S3_BUCKET="your-bucket-name"
```

File: docs/mkdocs.yml [hunk 2]
```diff
@@ -32,6 +32,7 @@ nav:
       - Media Backend:
           - MinIO: guides/minio-image-upload.md
           - S3: guides/s3-image-upload.md
+          - Google Cloud Storage Buckets: guides/s3-google-cloud-storage.md
```
Comment (ErikMichelson) [inline docs/mkdocs.yml:35]: That's shorter and precise enough

```suggestion
          - Google Cloud Storage: guides/s3-google-cloud-storage.md
```

**Review by ErikMichelson — CHANGES_REQUESTED** [review]
Sorry, this PR got lost in the amount of other PRs. 
Thanks for your contribution!

In general, it looks good, however as you're updating docs for current HedgeDoc 1.x, please use the HedgeDoc 1.x config environment variables.

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**